File: apps/mobile/lib/core/measurements/weight_processing_exception.dart
final class WeightProcessingException implements Exception {
  const WeightProcessingException(this.errorCode, this.message);

  static const rawRequired = 'WEIGHT_RAW_REQUIRED';
  static const rawInvalid = 'WEIGHT_RAW_INVALID';
  static const rawNegative = 'WEIGHT_RAW_NEGATIVE';
  static const rawIntegerDigitsExceeded = 'WEIGHT_RAW_INTEGER_DIGITS_EXCEEDED';
  static const rawFractionDigitsExceeded =
      'WEIGHT_RAW_FRACTION_DIGITS_EXCEEDED';
  static const decimalPlacesInvalid = 'WEIGHT_DECIMAL_PLACES_INVALID';
  static const processingMethodInvalid = 'WEIGHT_PROCESSING_METHOD_INVALID';

  final String errorCode;
  final String message;

  @override
  String toString() => 'WeightProcessingException($errorCode): $message';
}

File: services/backend/src/TraderPro.Domain/Common/Measurements/WeightProcessingException.cs
namespace TraderPro.Domain.Common.Measurements;

public sealed class WeightProcessingException : Exception
{
    public const string RawRequired = "WEIGHT_RAW_REQUIRED";
    public const string RawInvalid = "WEIGHT_RAW_INVALID";
    public const string RawNegative = "WEIGHT_RAW_NEGATIVE";
    public const string RawIntegerDigitsExceeded =
        "WEIGHT_RAW_INTEGER_DIGITS_EXCEEDED";
    public const string RawFractionDigitsExceeded =
        "WEIGHT_RAW_FRACTION_DIGITS_EXCEEDED";
    public const string DecimalPlacesInvalid =
        "WEIGHT_DECIMAL_PLACES_INVALID";
    public const string ProcessingMethodInvalid =
        "WEIGHT_PROCESSING_METHOD_INVALID";

    public WeightProcessingException(string errorCode, string message)
        : base(message)
    {
        ErrorCode = errorCode;
    }

    public string ErrorCode { get; }
}
